### PR TITLE
feat(url): Add valkey and valkeys schemes

### PR DESCRIPTION
**Pull request by jouir**
Inspired by the valkey-py client.

https://github.com/valkey-io/valkey-py/blob/v6.0.2/valkey/_parsers/url_parser.py#L40

### CODE REVIEW

**Comment by jouir** [comment]
Added tests

File: url_test.go [hunk 1]
```diff
@@ -15,12 +15,18 @@ func TestParseURL(t *testing.T) {
 	if opt, err := ParseURL("rediss://"); err != nil || opt.TLSConfig == nil {
 		t.Fatalf("unexpected %v %v", opt, err)
 	}
+	if opt, err := ParseURL("valkeyss://"); err != nil || opt.TLSConfig == nil {
```
Comment (rueian) [inline url_test.go]: Should this be valkeys?

Comment (jouir) [reply url_test.go]: You are right! It's fixed. Thanks.

**Comment by jouir** [comment]
I've also removed the test on the init address for the valkey scheme because the goal on this line is to test the scheme, not the init address that is already covered by other tests.

**Pull request merged**